staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go

```diff
@@ -115,48 +131,51 @@ type requestManagementSystem struct {
-	// freshly constructed map of priority level states.  But the new
-	// `RMState.priorityLevelStates` includes in its range at least all the
-	// `*PriorityLevelState` values of the old `RMState.priorityLevelStates`.
-	// Consequently the filter can load a `*RMState` and work with it
-	// without concern for concurrent updates.  When a priority level is
-	// finally deleted, this will also involve storing a new `*RMState`
-	// pointer here, but in this case the range of the
-	// `RMState.priorityLevels` will be reduced --- by removal of the
-	// priority level that is no longer in use.
+	// the name of global-default priority level
+	globalDefaultPriorityLevelConfigurationName string
+
+	// curState holds a pointer to the current requestManagementState.
+	// That is, `Load()` produces a `*requestManagementState`.  When a
+	// config work queue worker processes a configuration change, it
+	// stores a new pointer here --- it does NOT side-effect the old
+	// `requestManagementState` value.  The new
+	// `requestManagementState` has a freshly constructed slice of
+	// FlowSchema pointers and a freshly constructed map of priority
+	// level states.
 	curState atomic.Value
 }
 
 // NewRequestManagementSystem creates a new instance of request-management system
 func NewRequestManagementSystem(
 	informerFactory kubeinformers.SharedInformerFactory,
+	queueSetFactory fq.QueueSetFactory,
 	serverConcurrencyLimit int,
 	requestWaitLimit time.Duration,
-	clk clock.Clock,
 ) Interface {
 	reqMgmt := &requestManagementSystem{
-		clk:                    clk,
-		configQueue:            workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue"),
-		plLister:               informerFactory.Flowcontrol().V1alpha1().PriorityLevelConfigurations().Lister(),
-		fsLister:               informerFactory.Flowcontrol().V1alpha1().FlowSchemas().Lister(),
-		serverConcurrencyLimit: serverConcurrencyLimit,
-		requestWaitLimit:       requestWaitLimit,
+		queueSetFactory:                             queueSetFactory,
+		serverConcurrencyLimit:                      serverConcurrencyLimit,
+		requestWaitLimit:                            requestWaitLimit,
+		globalDefaultPriorityLevelConfigurationName: rmtypesv1alpha1.PriorityLevelConfigurationNameDefault,
 	}
-	// TODO: finish implementation
+	klog.V(2).Infof("NewRequestManagementSystem with serverConcurrencyLimit=%d, requestWaitLimit=%s", serverConcurrencyLimit, requestWaitLimit)
+	reqMgmt.initializeConfigController(informerFactory)
+	emptyRMState := &requestManagementState{
+		priorityLevelStates: make(map[string]*priorityLevelState),
+	}
+	reqMgmt.curState.Store(emptyRMState)
+	reqMgmt.digestConfigObjects(
+		[]*rmtypesv1alpha1.PriorityLevelConfiguration{
+			DefaultPriorityLevelConfigurationExempt,
+			DefaultPriorityLevelConfigurationDefault,
+		},
+		[]*rmtypesv1alpha1.FlowSchema{
+			DefaultFlowSchemaSystemTop,
```

Comment:
I see no attempt to preserve this item, so Wait has to cope with its absence.  Given that, why bother put it here?